lib/dialects/abstract/query-generator.js

```diff
@@ -2228,6 +2230,38 @@ const QueryGenerator = {
     return result ? result : '1=1';
   },
 
+  // A recursive parser for nested where conditions
+  parseConditionObject(conditions, path) {
+    path = path || [];
+    return _.reduce(conditions, (result, value, key) => {
+      if (_.isObject(value)) {
+        result = result.concat(this.parseConditionObject(value, path.concat(key))); // Recursively parse objects
+      } else {
+        result.push({ path: path.concat(key), value: value });
+      }
+      return result;
+    }, []);
+  },
+
+  /**
+   * Generates an SQL query that extract JSON property of given path.
+   *
+   * @param   {String}               column  The JSON column
+   * @param   {String|Array<String>} [path]  The path to extract (optional)
+   * @returns {String}                       The generated sql query
+   * @private
+   */
+  jsonPathExtractionQuery(column, path) {
+    const paths = _.toPath(path);
+    const pathStr = `{${paths.join(',')}}`;
+    const quotedColumn = this.isIdentifierQuoted(column) ? column : this.quoteIdentifier(column);
```

Comment:
If it called via `whereItemQuery`.  [abstract/query-generator.js#1945](https://github.com/sequelize/sequelize/pull/7094/files/dd8eba5554616b4793097d6f663664c1b71c70a3#diff-def52474024b12c7d133bb504cdf57d1R1949)
But not quoted when it called via Sequelize.json(). see[postgres/query-generator.js#211](https://github.com/sequelize/sequelize/pull/7094/files/dd8eba5554616b4793097d6f663664c1b71c70a3#diff-7e347c4738c97261969fbeb802a3c363R211)